cinema/tests/test_movie_session_api.py

```diff
@@ -0,0 +1,172 @@
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+
+from rest_framework.test import APIClient
+from rest_framework.reverse import reverse
+from rest_framework import status
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieListSerializer, MovieDetailSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+
+def detail_url(movie_id: int) -> str:
+    return reverse("cinema:movie-detail", args=(movie_id,))
+
+
+def sample_movie(**params) -> Movie:
+    defaults = {
+        "title": "Groundhog Day",
+        "description": "A narcissistic, self-centered weatherman finds himself in a time loop on Groundhog Day.",
+        "duration": "101",
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+class UnauthenticatedCinemaApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedCinemaApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test_user@test.com",
+            password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movie_list(self):
+        sample_movie()
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+        res = self.client.get(MOVIE_URL)
+        self.assertTrue(res.status_code, status.HTTP_200_OK)
```

Comment:
The assertion `self.assertTrue(res.status_code, status.HTTP_200_OK)` should be `self.assertEqual(res.status_code, status.HTTP_200_OK)` to correctly compare the response status code.